### PR TITLE
Add test for the finalized blocks with the comparePopScore

### PR DESCRIPTION
**Pull request by Mr-Leshiy**

### CODE REVIEW

File: test/pop/blockchain/altblock_finalization_test.cpp [hunk 1]
```diff
@@ -508,4 +508,29 @@ TEST_F(AltBlockFinalization2, FinalizeMaxBtcs) {
...
 
   ASSERT_TRUE(alttree.setState(*tip, state));
+}
+
+TEST_F(AltBlockFinalization2, comparePopScore_test) {
+  altparam.mEndorsementSettlementInterval = 10;
+  altparam.mPreserveBlocksBehindFinal = 10;
+
+  // mine 10 altblocks
+  auto *forkPoint = mineAltBlocks(*alttree.getBestChain().tip(), 50);
```
Comment (blackyblack) [inline test/pop/blockchain/altblock_finalization_test.cpp:518]: Looks like you mine 50 blocks not 10.
And what is this test case about? Make some intro.

Comment (Mr-Leshiy) [reply test/pop/blockchain/altblock_finalization_test.cpp:526]: Fixed

File: test/pop/blockchain/altblock_finalization_test.cpp [hunk 2]
```diff
@@ -508,4 +508,37 @@ TEST_F(AltBlockFinalization2, FinalizeMaxBtcs) {
   assertTreesHaveNoOrphans(alttree);
 
   ASSERT_TRUE(alttree.setState(*tip, state));
+}
+
+TEST_F(AltBlockFinalization2, comparePopScore_test) {
+  // Test scenario:
+  // mine 50 alt blocks
+  // creating to chains: chainA (56 height), chainB(55 height)
```
Comment (blackyblack) [inline test/pop/blockchain/altblock_finalization_test.cpp:516]: two

File: test/pop/blockchain/altblock_finalization_test.cpp [hunk 3]
```diff
@@ -508,4 +508,37 @@ TEST_F(AltBlockFinalization2, FinalizeMaxBtcs) {
...
+  // active chain should be on chainA
+  // saving trees in the storage
```
Comment (blackyblack) [inline test/pop/blockchain/altblock_finalization_test.cpp:518]: to the

File: test/pop/blockchain/altblock_finalization_test.cpp [hunk 4]
```diff
@@ -508,4 +508,37 @@ TEST_F(AltBlockFinalization2, FinalizeMaxBtcs) {
...
+  // finalizing the tip of the chainA, so should be removed chainB
```
Comment (blackyblack) [inline test/pop/blockchain/altblock_finalization_test.cpp:519]: chainB should be removed

File: test/pop/blockchain/altblock_finalization_test.cpp [hunk 5]
```diff
@@ -508,4 +508,37 @@ TEST_F(AltBlockFinalization2, FinalizeMaxBtcs) {
...
+  // trying to compare chainA with the deleted chainB, chainA should wins
```
Comment (blackyblack) [inline test/pop/blockchain/altblock_finalization_test.cpp:520]: should win

**Review by blackyblack — APPROVED** [review]

**Review by blackyblack — APPROVED** [review]

**Pull request merged**